libc/test/src/__support/freelist_heap_test.cpp

```diff
@@ -47,10 +48,16 @@ TEST_FOR_EACH_ALLOCATOR(CanAllocate, 2048) {
   void *ptr = allocator.allocate(ALLOC_SIZE);
 
   ASSERT_NE(ptr, static_cast<void *>(nullptr));
-  // In this case, the allocator should be returning us the start of the chunk.
-  EXPECT_EQ(ptr, static_cast<void *>(
-                     reinterpret_cast<cpp::byte *>(allocator.region_start()) +
-                     FreeListHeap<>::BlockType::BLOCK_OVERHEAD));
```

Comment:
From the changes in this patch, this actually isn't always guaranteed now even for a fresh allocator since the new `MIN_ALLOCATION` is dependent on `max_align_t` which could be larger than the alignment for `BlockType`. In this case, the allocation may no longer be guaranteed to be the very first block.

Reply:
can't this test control the allocator config, such that you can guarantee that invariant? If you think the test isn't useful, that's a different problem, but the rationale you've described sounds like it should be controllable in the test setup.

Reply:
From the allocator perspective, I think it shouldn't matter where the actual block is located. We could configure an allocator object such that this is guaranteed for the very first block, but I think that might be trying to test the underlying machinery rather than the allocator API which doesn't really guarantee it.

Reply:
OK, so we're in the scenario, where you don't think the test was valuable. Thanks for the clarification. I was operating under the impression, that the test was viewed as useful, but not stable enough to keep.